release-notes/aws/v9.2.2.md

```diff
@@ -0,0 +1,37 @@
+## :zap: Giant Swarm Release 9.2.2 for AWS is now active for you! :zap:
+
+**If you are upgrading from 9.2.1, upgrading to this release merely updates the apps. It will *not* roll your nodes.**
+
+**It is not recommended to upgrade from 9.0.0 or older platform release, please only upgrade to 9.2.2 from 9.0.1 or newer once they are available.**
+
+Rationale for this is that 9.0.0 and older platform releases include apps which are still using Kubernetes 1.16 incompatible APIs
```

Comment:
I agree with Ross. And can we also add the value that this release brings? e.g. 

---

This release makes preinstalled apps on 9.x.x releases compatible with Kubernetes 1.16. It creates a safe path to upgrade from Kubernetes 1.15 to 1.16.

Because 9.0.0 and older platform releases contain apps which use Kubernetes 1.16 incompatible APIs, please note:

**It is not recommended to upgrade from 9.0.0 or older platform releases. Please only upgrade to 9.2.2 from 9.0.1 or newer once they are available.**

---

WDYT? I just want to explain the value of this release.

Reply:
> This release makes preinstalled apps on 9.x.x releases compatible with Kubernetes 1.16. It creates a safe path to upgrade from Kubernetes 1.15 to 1.16.

This is not true, apps were already compatible with k8s 1.16, in aws 9.2.1 if not earlier. In 9.2.2 we just update some of the apps to include their bug fixes, upstream binaries and our charts, but there are no k8s 1.16 compatibility fixes there at least that I'm aware of which weren't already in 9.2.1.

Apps in 9.0.0 aren't compatible with k8s 1.16, and due to how our tenant cluster upgrade process is implemented (it upgrades tenant cluster's k8s master first before updating apps) apps can end up in broken state - some k8s 1.15 API references (specifically state helm stores) from previous app installation (from 9.0.0 platform release) can remain not replaced entirely - and this is true whether one is upgrading from 9.0.0 to 9.2.1 or to 9.2.2.

We do not include in 9.2.2 a fix for upgrade logic that would take care to upgrade apps appropriately when upgrading from 9.0.0 or older, therefore we have note about not recommending upgrades to 9.2.2 from 9.0.0 or older, 9.2.2 doesn't create safe path to upgrade from 1.15 based releases like 9.0.1

Even if 9.2.2 included such fix for cluster upgrade logic, it wouldn't help to clusters which are on 9.2.1 already, since app updates there are in broken state - to recover those we have to manually intervene with hard down time for those affected apps in affected clusters, to delete the broken apps and have them reinstalled; idea was to do this app deletion + re-installation during upgrade from 9.2.1 to 9.2.2, assuming app downtime would be more acceptable and getting maintenance window easier, given incentive to get some bug fixes in the process too; challenge is to communicate that problem and solution in release notes; @rossf7 was proposing not to do it at all in release notes, instead have SEs bring it to customers with affected clusters or we could write something down that could be shared with those customers only as follow up to 9.2.2 announcement; if we releases 9.0.1 together with 9.2.2, that should make it impossible to get more affected clusters with app update stuck, at least for k8s 1.15 to 1.16 upgrade for legacy aws clusters; I'm curious what @pipo02mix thinks about that approach?

Reply:
@puja108 could you help me here? Do you think the original main paragraph would be enough to communicate what's in this release? It's pretty tough to explain

Reply:
I think we can simplify it radically, now that 9.0.1 is out; again assuming info about need to recover apps by reinstalling them gets delivered directly/separately to ones that weren't so lucky, didn't have 9.0.1, so ended up upgrading to 9.2.1 already from 9.0.0.

So, leave out all the stuff about safe path, there's no more workaround skipping 9.0.1 if regular upgrade procedures gets followed.

Reply:
This is getting very confusing now, it used to be a comment about not upgrading from a certain version and now it is a discussion about value. But I'll try to untangle this now and come up with sth

Reply:
9.0.1 is out, 9.2.2 release notes have been simplified, and aligned with https://github.com/giantswarm/releases/pull/203

PTAL

Reply:
Resolved w Stevo's simplified change to main paragraph.